### PR TITLE
[5.0] Normalize speculative & producer block intervals

### LINKED ISSUE #1837: 5.0: lots of "not producing block" messages on the backup BP
With 5.0 (latest), we get a lot of "not producing block" messages on the backup BP.  The backup BP has the wrong key on purpose.

In previous version we just get 1 of these per block.

```
$ grep 'Block #105321099 eosnationftw' /var/log/syslog | wc -l
763
```

image

### PR DESCRIPTION
**Pull request by heifner**
#1481 changed speculative block creation so that speculative blocks would always have a current block time. However, it didn't correctly handle the case where the configured producer was the scheduled producer but could not produce (e.g. incorrect configured private key). This PR simplifies the implementation to just use the producer schedule as configured by `produce-block-offset-ms` regardless if the block is speculative or a producer block. This normalizes block production time for speculative and producer blocks. I think this normalization is better than maintaining different block intervals for speculative vs producer nodes.

Resolves #1837

### CODE REVIEW

**Review by linh2931 — COMMENTED** [review]
Is it possible to add a test where the scheduled producer could not produce?

**Comment by heifner** [comment]
> Is it possible to add a test where the scheduled producer could not produce?

I think that would be under https://github.com/AntelopeIO/leap/issues/1804

There are tests in `test_block_timing_util.cpp` that test the timings, so I feel the functionality is well tested; just not the actual usage here.

**Comment by linh2931** [comment]
> I think that would be under #1804
> 
> There are tests in `test_block_timing_util.cpp` that test the timings, so I feel the functionality is well tested; just not the actual usage here.

I meant the situation reported by #1837. No more lots of "not producing" messages after this fix. But if it is too much effort, there is no need to do that, as the fix makes things simpler. I am approving the PR.

**Review by linh2931 — APPROVED** [review]

File: plugins/producer_plugin/producer_plugin.cpp [hunk 1]
```diff
@@ -1830,7 +1830,7 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
    }
 
    // create speculative blocks at regular intervals, so we create blocks with "current" block time
-   _pending_block_deadline = now + fc::microseconds(config::block_interval_us);
+   _pending_block_deadline = block_timing_util::calculate_producing_block_deadline(_produce_block_cpu_effort, block_time);
```
Comment (greg7mdp) [inline plugins/producer_plugin/producer_plugin.cpp:1833]: I believe that `calculate_producing_block_deadline` will return `std::nullopt` if `producers.empty()`.
Is it OK to not have a deadline in that case?

Comment (heifner) [reply plugins/producer_plugin/producer_plugin.cpp:1833]: This is calling `calculate_producing_block_deadline` not `calculate_producer_wake_up_time`

Comment (greg7mdp) [reply plugins/producer_plugin/producer_plugin.cpp:1833]: Oh, my bad, thanks!

**Review by greg7mdp — APPROVED** [review]

**Pull request merged**